### DISCUSSION TITLE
No ACK packets from client

### DISCUSSION THREAD

**Discussion by JYBESiU** [ANLAB-KAIST/KENSv3#187] [Project Q&A 2. Transport Layer (TCP)]
I am implementing 3-way handshaking.
<img width="1651" alt="스크린샷 2024-04-08 오후 7 41 05" src="https://github.com/ANLAB-KAIST/KENSv3/assets/39625161/a3561882-65dd-46e1-95bf-c1a988e4a86d">

When the client send SYN to the server, server send SYN-ACK to the client correctly.
But the client doesn't resend a ACK to the server.
I implemented sending ACK packets behavior in `packetArrived` function.

What can be the possible reasons for this situation?

**Comment by Pusnow** [comment]
It seems your SYN-ACK packets are dropped. There are various reasons for it.

Please, try turning on debug flags in NetworkLog especially `TCP_LOG` to find where the packets are dropped.

https://github.com/ANLAB-KAIST/KENSv3/blob/5bf97d4d228d6a6bcb8a2fdc1f50abe9c075c701/src/Networking/E_NetworkLog.cpp#L13-L25
